src/core/server/http/http_server.ts

```diff
@@ -133,19 +134,26 @@ export class HttpServer {
     }
   }
 
-  public async setup(config: HttpConfig): Promise<HttpServerSetup> {
+  public async setup(
+    config: HttpConfig,
+    executionContext?: InternalExecutionContextSetup
+  ): Promise<HttpServerSetup> {
     const serverOptions = getServerOptions(config);
     const listenerOptions = getListenerOptions(config);
     this.server = createServer(serverOptions, listenerOptions);
     await this.server.register([HapiStaticFiles]);
     this.config = config;
 
+    // It's important to have setupRequestStateAssignment call the very first, otherwise context passing will be broken.
+    // That's the only reason why context initialization exists in this method.
+    this.setupRequestStateAssignment(config, executionContext);
```

Comment:
That's the only reason I had to declare HTTP server dependency on execution context service. I need to test whether an implementation based on the APM agent will have the same limitation.